### PR TITLE
Unblock dropwizard and test infra dependencies

### PR DESCRIPTION
**Pull request by ergo14**
## General
**Before this PR**:
Bumps of Junit and Mockito, see `versions.lock` comments for details. This means excavators are not able to upgrade some of our dependencies (e.g. it bumps Junit transitively).

**After this PR**:
Bump dropwizard as a first step to unblock.
==COMMIT_MSG==
==COMMIT_MSG==

**Priority**:
P1
**Concerns / possible downsides (what feedback would you like?)**:
See PR comments
**Is documentation needed?**:
No
## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:
No
**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:
No
**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:
Yes
**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:
No
**Does this PR need a schema migration?**
No
## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:
TimeLock partitioning and single-leader batching are not enabled anywhere.
**What was existing testing like? What have you done to improve it?**:

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:
N/A
**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:
N/A
## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:
No-op - I can bump the other dependencies in peace
**Has the safety of all log arguments been decided correctly?**:
N/A
**Will this change significantly affect our spending on metrics or logs?**:
No
**How would I tell that this PR does not work in production? (monitors, etc.)**:
No op
**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:
Rollback
**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:
No
**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:
No
**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@raiju

### CODE REVIEW

File: atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java [hunk 1]
```diff
@@ -1518,7 +1518,7 @@ private SingleBatchSweeper getSingleShardSweeper(SettableFuture<MultiTableSweepQ
      */
     private TimelockService createStickyLockService() {
         AtomicLong lockToken = new AtomicLong(0);
-        Set<LockDescriptor> requestedLocks = new ConcurrentHashSet<>();
```
Comment (ergo14) [inline atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java:1521]: Got deprecated

File: lock-impl/src/test/java/com/palantir/lock/impl/LockServiceImplTest.java [hunk 2]
```diff
@@ -1,195 +0,0 @@
...
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
-import uk.org.lidalia.slf4jext.Level;
-import uk.org.lidalia.slf4jtest.LoggingEvent;
-import uk.org.lidalia.slf4jtest.TestLogger;
-import uk.org.lidalia.slf4jtest.TestLoggerFactory;
-
-public final class LockServiceImplTest {
```
Comment (ergo14) [inline lock-impl/src/test/java/com/palantir/lock/impl/LockServiceImplTest.java:54]: See internal slack thread on the BLUF:
- Bumping dropwizard ending up pulling a newer version of slf4j API
- slf4j-test does not implement that version
- Tests end up not loading any loggers for the reasons above and the tests fail
- It is possible to hack on something to keep this test, but giving (1) testing logging is rare, (2) the class tested is legacy and something we rarely touch and want to deprecate and (3) it's more tech debt. I decided it's best we unblock deps and remove the test altogether

File: timelock-server/src/integ2Test/java/com/palantir/atlasdb/timelock/AbstractSingleLeaderMultiNodePaxosTimeLockIntegrationTest.java [hunk 3]
```diff
@@ -108,6 +109,9 @@ public void migrationToBatchedSingleLeaderHasConsistentSequenceNumbers() {
         assertThat(ImmutableSet.copyOf(sequenceNumbers)).hasSameSizeAs(sequenceNumbers);
     }
 
+    // not fixing, batched paxos (for timestamps, for multi leader and for single leader) is not used
+    // and will be removed soon in the future
+    @Disabled
     @Test
```
Comment (ergo14) [inline timelock-server/src/integ2Test/java/com/palantir/atlasdb/timelock/AbstractSingleLeaderMultiNodePaxosTimeLockIntegrationTest.java]: It seems like the 3 iteration end up accumulating 3 values where 2 are duplicated which is weird, and not sure how this ends up happening

These tests test batching/timelock-partitioning which we do not use and want to remove. Thus, not investing more time to understand and fix

Feel free to spend a couple of cycles to see if you can figure this out

File: versions.props [hunk 4]
```diff
@@ -84,7 +84,7 @@ uk.org.lidalia:slf4j-test = 1.1.0
 com.palantir.docker.compose:* = 1.9.0
 
 # Pinning Dropwizard 2.1.x until we can upgrade tests
-io.dropwizard:* = 2.1.7
+io.dropwizard:* = 3.0.0
```
Comment (ergo14) [inline versions.props]: I am going to update this comment before merging to
- Fix the version referenced in the comment
- Leave a to-do to remove this from the pinned section once all is resolved

File: timelock-server/src/integ2Test/java/com/palantir/atlasdb/timelock/AbstractSingleLeaderMultiNodePaxosTimeLockIntegrationTest.java (hunk 3 above)
Comment (ergo14) [reply timelock-server/src/integ2Test/java/com/palantir/atlasdb/timelock/AbstractSingleLeaderMultiNodePaxosTimeLockIntegrationTest.java]: I spent a bit more time reading the test setup and where dropwizard is used but did not find anything suspect.

Though, I found that the opposite symmetric this (above this one) has an additional if statement. I added it and the test passes now! I have no idea why it stopped working on this upgrade and was passing before (I re-triggered CI 3 times when it was failing, and checked it was consistently failing locally).

I think this is OK as-is given we throw these tests anyway. I am no longer worried about the upgrade messing up something in TestableTimelockServer

File: lock-impl/src/test/java/com/palantir/lock/impl/LockServiceImplTest.java [hunk 5]
```diff
@@ -1,195 +0,0 @@
...
-    @Test
-    public void debugOrSlowLogShouldNotBeLoggedWhenLockResponseIsNotSlow() {
-        lockServiceWithSlowLogEnabled.logSlowLockAcquisition(
-                TEST_LOCK, LockClient.ANONYMOUS, LockServiceImpl.DEBUG_SLOW_LOG_TRIGGER_MILLIS - 5);
-        assertThat(testLockServiceImplLogger.getLoggingEvents()).isEmpty();
-        assertThat(testSlowLogger.getLoggingEvents()).isEmpty();
-    }
-
-    @Test
-    public void verifySerializedBatchOfLockRequestsSmallerThan45MB() throws InterruptedException, IOException {
```
Comment (mdaudali) [inline lock-impl/src/test/java/com/palantir/lock/impl/LockServiceImplTest.java:167]: keep this one I think, although I'm surprised that all the other tests in this class are just log related!

**Review by mdaudali — APPROVED** [review]
Thanks! Make the one change noted above, but lgtm.

**Pull request merged**